packages/block-library/src/index.js

```diff
@@ -145,3 +153,25 @@ export const registerCoreBlocks = () => {
 		setGroupingBlockName( group.name );
 	}
 };
+
+/**
+ * Function to register experimental core blocks depending on editor settings.
+ *
+ * @param {Object} settings     Editor settings.
+ *
+ * @example
+ * ```js
+ * import { registerExperimentalCoreBlocks } from '@wordpress/block-library';
+ *
+ * registerExperimentalCoreBlocks( settings );
+ * ```
+ */
+export const registerExperimentalCoreBlocks = process.env.GUTENBERG_PHASE === 2 ? ( settings ) => {
```

Comment:
Shouldn't we prefix it with `__experimental` as this is not meant to be a part of the public API?

See: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#experimental-and-unstable-apis